aries_cloudagent/revocation/manager.py

```diff
@@ -147,7 +151,27 @@ async def revoke_credential(
                 await txn.commit()
             await self.set_cred_revoked_state(rev_reg_id, crids)
             if delta_json:
-                await issuer_rr_upd.send_entry(self._profile)
+                if write_ledger:
+                    await issuer_rr_upd.send_entry(self._profile)
+                else:
+                    async with self._profile.session() as session:
+                        try:
+                            connection_record = await ConnRecord.retrieve_by_id(
+                                session, connection_id
+                            )
+                        except StorageNotFoundError:
+                            raise RevocationManagerError(
+                                f"No connection record found for id: {connection_id}"
```

Comment:
Is this checking to see if there is an endorser connection?  Is so, the error message should be more explicit, I think.

Reply:
Yes, to check that the `connection_id` if provided when calling `POST /revocation/revoke` exists. If no `connection_id` is provided and the profile is author role then connection_id will be retrieved from `endorser.endorser_alias` which will always exist. Maybe I can change it to `No endorser connection record .... `